modAionImpl/src/org/aion/zero/impl/sync/handler/BlockPropagationHandler.java

```diff
@@ -157,7 +163,10 @@ public PropStatus processIncomingBlock(
             }
         } else {
             result = this.blockchain.tryToConnect(block);
+
             long t2 = System.currentTimeMillis();
+            this.syncStats.updatePeerImportedBlocks(_displayId, 1);
```

Comment:
1. Before updating the stats you need to check that the result indicates storage.
2. The stats also need to be updated above where we attempt to store the block and the result is successful.
3. The blocks received by this class should also be counted towards the total number of received blocks.